### DISCUSSION TITLE
Get specific table size on disk in (bytes, KB, MB, GB, etc)

### DISCUSSION THREAD

**Discussion by kevinschaich** [duckdb/duckdb#12117] [Q&A]
We're using `PRAGMA database_size` to get the size on disk for our entire database.

Is there a similar operation to get the size on disk for a specific table?

We tried `PRAGMA storage_info('table_name');` but that doesn't seem to be exactly what we're looking for

**Comment by renbud** [comment]
I think you can use the number of distinct block_id from pragma_storage_info combined with the block_size from pragma_database_size.  The numbers look close to me if I verify the total against the total database size.

To get info for {{mytable}}

`select {{mytable}} as table_name, count(distinct block_id) as num_blocks, count(distinct block_id) * (select block_size from pragma_database_size()) as num_bytes from pragma_storage_info({{mytable}}) group by all`

**Comment by thenicekat** [comment]
Did we figure anything out here?
